app/models/institution.rb

```diff
@@ -0,0 +1,42 @@
+class Institution < ApplicationRecord
+  attr_accessor :banko
+
+  def banko
+    @banko ||= Money::Bank::VariableExchange.new(self)
+  end
+
+  def rates
+    @rates ||= ExchangeRate.where(institution_id: self.id)
+  end
+
+  def set_bank
+    Money.default_bank = banko
+  end
+
+  def get_amount(from_iso_code, to_iso_code, from_amount)
+    set_bank
+    Money.from_amount(from_amount, from_iso_code).exchange_to(to_iso_code)
+  rescue Money::Bank::UnknownRate => e
+    nil
+  end
+
+  #
+  # get_rate is exposed to be accessible to Money::Bank::VariableExchange
+  # it's not supposed to be accessed directly.
```

Comment:
should encapsulate under private.
If not possible for Money gem, move this to another class

same for `get_rate`